### ISSUE TITLE
Test case: history_010_pos

### ISSUE THREAD

**Issue by bunder2015** [zfsonlinux/zfs#6301]
### System information

Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       | Gentoo
Distribution Version    | n/a
Linux Kernel                 | 4.9.16
Architecture                 | amd64
ZFS Version                  | head 0.7.0-rc4_86_gfe46eebe6
SPL Version                  | head 0.7.0-rc4_5_g7a35f2b

### Describe the problem you're observing
Temporary user created to run test can't find the zfs binary
### Describe how to reproduce the problem
Run tests
### Include any warning/errors/backtraces from the system logs
```
Test: /home/testuser/zfs/tests/zfs-tests/tests/functional/history/history_010_pos (run as root) [00:00] [FAIL]
23:24:44.82 ASSERTION: Verify internal long history information are correct.
23:24:44.87 SUCCESS: useradd -g hgroup -d /var/tmp/huser -m huser
23:24:44.90 SUCCESS: usermod -a -G testuser huser
23:24:44.97 SUCCESS: eval zfs create testpool/testfs1
23:24:45.00 SUCCESS: verify_long zfs create testpool/testfs1 root -l
23:24:45.04 SUCCESS: eval zfs allow hgroup snapshot,mount testpool/testfs1
23:24:45.07 SUCCESS: verify_long zfs allow hgroup snapshot,mount testpool/testfs1 root -l
23:24:45.11 SUCCESS: eval zfs allow huser destroy,mount testpool/testfs1
23:24:45.14 SUCCESS: verify_long zfs allow huser destroy,mount testpool/testfs1 root -l
23:24:45.18 SUCCESS: eval zfs allow huser reservation testpool/testfs1
23:24:45.21 SUCCESS: verify_long zfs allow huser reservation testpool/testfs1 root -l
23:24:45.25 SUCCESS: eval zfs allow huser allow testpool/testfs1
23:24:45.27 SUCCESS: verify_long zfs allow huser allow testpool/testfs1 root -l
23:24:45.30 ERROR: su huser -c eval zfs snapshot testpool/testfs1@snap exited 127
23:24:45.30 bash: zfs: command not found
```

**Comment by MyPod** [comment]
this happens because /usr/sbin is not in the user's PATH

**Comment by bunder2015** [comment]
When running the tests, the binaries are run from the current build eg: ./zfs/cmd/zfs/zfs ./zfs/cmd/zpool/zpool etc.

**Comment by gmelikov** [comment]
@bunder2015 IIRC no, special folder with symlinks to every needed command is created, and it must be added to PATH, `23:24:45.30 ERROR: su huser -c eval zfs snapshot testpool/testfs1@snap exited 127` doesn't have it for some reason (su doesn't get it for some reason). I think it's distributive related (gentoo), because other distros in buildbot doesn't have such problem, see https://github.com/zfsonlinux/zfs/pull/5903#issuecomment-291631633

**Comment by bunder2015** [comment]
Thanks for pointing me in the right direction.  I changed https://github.com/zfsonlinux/zfs/blob/master/tests/zfs-tests/tests/functional/history/history_common.kshlib#L77 to `su - $user ...` and all is well.  I'll submit a patch.

**Comment by bunder2015** [comment]
Closing, will be fixed in #6303
